src/services/sign.js

```diff
@@ -1,22 +1,29 @@
 // eslint-disable-next-line import/extensions
 import {Buffer} from 'buffer';
 import {sha3} from './hashUtils/sha3';
+import {removeFromAsyncStorage, setToAsyncStorage} from '../actions/utils';
 
 global.Buffer = Buffer;
 
 // const Buffer = require('buffer').Buffer;
 const EthereumTx = require('ethereumjs-tx').Transaction;
 const formatters = require('web3-core-helpers').formatters;
+const keythereum = require('keythereum');
+
+// chainList for ethereumjs-tx
+const chainList = {
+  1: 'mainnet',
+  3: 'ropsten',
+  4: 'rinkeby',
+  5: 'goerli',
+};
 
 // sign an unsigned raw transaction and deploy
-function deployUnsignedTx(tx: any, privateKey?: any, testnetId?: any) {
-  // TODO: error handling
+export function deployUnsignedTx(tx: any, privateKey?: any, testnetId?: any) {
   try {
-    // tx = JSON.parse(tx);
     const txData = formatters.inputTransactionFormatter(tx);
-
     // TODO: this method should not work for ganache and prysm and throw error
-    const chainId = Number(testnetId) === 5 ? 6284 : Number(testnetId);
```

Comment:
`6284` works in ethcode. https://github.com/quantanet/ethcode/blob/b97d092d24fa470c0de323bdde729e39097bbb6f/ext-src/worker.ts#L115
How this changed ?